src/flow_graph.py

```diff
@@ -1096,15 +1274,163 @@ def reset_block_info(self) -> None:
             node.block.block_info = None
 
 
+def phi_reg_sources(node: Node, reg: Access, imdom_srcs: RefSet) -> RefSet:
+    """
+    Return the RefSet of all of the places `reg` is assigned to, if it is a valid phi.
+    Otherwise, return RefSet.invalid().
+    """
+    if node.immediate_dominator is None:
+        return RefSet.invalid()
+
+    seen = {node.immediate_dominator}
+    stack = node.parents[:]
+    sources = RefSet()
+
+    while stack:
+        n = stack.pop()
+        if n == node.immediate_dominator and not imdom_srcs.is_valid():
+            return RefSet.invalid()
```

Comment:
if we reach immediate_dominator, don't we want to merge imdom_srcs into sources? (rather than having the `sources.is_valid()` check at the end of this function, which misses cases)